src/types/fieldExtractor.ts

```diff
@@ -81,3 +126,19 @@ export const resolveEthereumTrackIdOverrides = (nft: NFT, contract: NftFactory):
   }
   return ethereumTrackId(nft.contractAddress, trackId);
 }
+
+export const resolveArtistNameOverrides = (nft: NFT, contract: NftFactory): string => {
+  const artistNameExtractorOverride = contract.typeMetadata?.overrides?.extractor?.artistName;
+  if (!artistNameExtractorOverride || (artistNameExtractorOverride !== ArtistNameExtractorTypes.ATTRIBUTES_TRAIT_MUSICIAN)) {
+    return contract.platformId;
+  }
+  return artistNameExtractor(contract)(nft);
+}
+
+export const resolveArtistIdOverrides = (nft: NFT, contract: NftFactory): string => {
+  const artistIdExtractorOverride = contract.typeMetadata?.overrides?.extractor?.artistId;
+  if (!artistIdExtractorOverride || (artistIdExtractorOverride !== ArtistIdExtractorTypes.USE_PLATFORM_AND_ARTIST_NAME)) {
+    return contract.platformId;
+  }
```

Comment:
hm although this was the default before, i don't think it's a good default. i think we should just hard error here and make sure the legacy migrations include a USE_PLATFORM_ID or similar extractor

Reply:
we can also just manually update the legacy changed NftFactories so we don't have to do a rebuild

Reply:
I've changed this and am reviewing old migrations.

Question @musnit : if we change previous ones to `USE_PLATFORM_AND_ARTIST_NAME`, will that break any fav'd artists for existing users?

Reply:
probably yes but may not be a big deal.

having said that, ideally i don't think we should ever really be using platforms as part of the artist identity, since they will often exist across multiple platforms, so i wouldn't change to that. it's just a band-aid for relics for now i think

Reply:
Ok, having something like `ARTIST_NAME` probably makes more sense.

Reply:
I've made it fall back to the artist ID override, and it throws an error if no extract is provided and there's no artist ID override.